docker/docker_image_src.go

```diff
@@ -78,18 +79,19 @@ func (s *dockerImageSource) GetManifest(mimetypes []string) ([]byte, string, err
 	return manblob, res.Header.Get("Content-Type"), nil
 }
 
-func (s *dockerImageSource) GetBlob(digest string) (io.ReadCloser, error) {
+func (s *dockerImageSource) GetBlob(digest string) (io.ReadCloser, int64, error) {
 	url := fmt.Sprintf(blobsURL, s.ref.RemoteName(), digest)
 	logrus.Infof("Downloading %s", url)
 	res, err := s.c.makeRequest("GET", url, nil, nil)
 	if err != nil {
-		return nil, err
+		return nil, 0, err
 	}
 	if res.StatusCode != http.StatusOK {
 		// print url also
-		return nil, fmt.Errorf("Invalid status code returned when fetching blob %d", res.StatusCode)
+		return nil, 0, fmt.Errorf("Invalid status code returned when fetching blob %d", res.StatusCode)
 	}
-	return res.Body, nil
+	size, _ := strconv.ParseInt(res.Header.Get("Content-Lenght"), 10, 64)
```

Comment:
- Typo in header name! (the urgency of having working tests for all of this is increasing… I do need to get that done.)
- `strconv.ParseInt` does not _always_ return 0 on error. (I don’t really care whether we would fail or just return ”unknown” on failure, but silently returning a different value on overflow is not great.)